### ISSUE TITLE
Epic - ListViewSubItems in the ListView have incorrect behavior

### ISSUE THREAD

**Issue by SergeySmirnov-Akvelon** [dotnet/winforms#4690]
<details>
<summary> Common</summary>

### Case 1:
- [x] The checkbox does not have its accessibility object in the `ListView`

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111449792-c7757900-8720-11eb-914f-fd02d6b349ab.png)

**Expected behavior:**
The checkbox should have its accessibility object in the `ListView`
</details>

<details>
<summary> Details </summary>

### Case 1 (regression from #3223):
- [x] A yellow rectangle is drawn for the entire `ListViewItem` instead of the selected `ListViewSubItem`

**Actual behavior:**
![Issue-4690-details-case1](https://user-images.githubusercontent.com/23376742/111447695-ba578a80-871e-11eb-9dc4-39aa45238af1.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111448495-85980300-871f-11eb-9346-6d23b052b604.png)

### Case 2:
- [x] The image does not have its accessibility object in the `ListView`

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111451977-0f959b00-8723-11eb-8225-6a1551022007.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111452076-2cca6980-8723-11eb-837f-8b548e2bfa51.png)

### Case 3:
- [x] ListViewSubItem of an ListViewItem are not displayed in the Inspector tree when the ListViewItem is in a ListViewGroup (may be fixed in #4678)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111466674-a4ed5b00-8734-11eb-8ff2-d8dd253a49ed.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111466706-acacff80-8734-11eb-9f97-63d990548f91.png)
</details>

<details>
<summary> LargeIcon</summary>

### Case 1:
- [x] `ListViewSubItem` in a `ListView` do not have accessibility objects (https://github.com/dotnet/winforms/pull/8022)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111459193-6d2de580-872b-11eb-8b34-c26e8d6e9363.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111459197-71f29980-872b-11eb-8051-675e8a0f7ff4.png)
</details>

<details>
<summary> List</summary>

### Case 1:
- [x] `ListViewSubItem` in a `ListView` do not have accessibility objects (https://github.com/dotnet/winforms/pull/8022)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111455530-f7277f80-8726-11eb-93e7-58f8567ae2a9.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111455558-00185100-8727-11eb-8ca0-efc2831d156c.png)
</details>

<details>
<summary> SmallIcon</summary>

### Case 1:
- [x] `ListViewSubItem` in a `ListView` do not have accessibility objects (https://github.com/dotnet/winforms/pull/8022)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111458355-62268580-872a-11eb-8fb6-3c3ea56a2de9.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111458398-6c488400-872a-11eb-9211-83c4029e33bc.png)
</details>

<details>
<summary> Tile </summary>

### Case 1 (regression from #3223):
 - [x] `ListViewSubItem` of an `ListViewItem` are not displayed in the Inspector tree when the `ListViewItem` is in a `ListViewGroup` (#4678)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111444647-a4949600-871b-11eb-8381-817028884fa8.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111444728-bb3aed00-871b-11eb-8420-1f630c8f9468.png)

### Case 2 (regression from #3223):
 - [x] The rectangle is not displayed for a `ListViewSubItem` when the user hovers over it or selects in the Inspector tree (#4679)

**Actual behavior:**
![image](https://user-images.githubusercontent.com/23376742/111445629-a4e16100-871c-11eb-80fa-2e0502ad807f.png)

**Expected behavior:**
![image](https://user-images.githubusercontent.com/23376742/111445672-b1fe5000-871c-11eb-85a7-163805e7ad14.png)
</details>

**Comment by dkazennov** [comment]
I'll take Common.Case1

**Comment by dkazennov** [comment]
ListViewItem components of the ListView control have Accessible Objects. 
But the main issue as I see it: accessible tools do not fix checking the checkboxes of ListViewItems. The Narrator doesn't announces checks.  ListViewItem can contain checkbox, but it doesn't behave like CheckBox control. So the ListViewItem class must be changed, maybe new AccessibleObjects can be introduced (like in the MonthCalendar control).

**Comment by Tanya-Solyanik** [comment]
Only a single task is fixed so far.

**Comment by MandiMan** [comment]
Verified this issue in the latest .Net 8.0 SDK build:8.0.100-alpha.1.22478.15, it was fixed. please check below screenshot:
![image](https://user-images.githubusercontent.com/108860782/192973789-30a59069-c6de-4300-b7c1-39022706ee72.png)

**Comment by NikitaSemenovAkvelon** [comment]
Is it makes sense to add `ListViewItemImageAccessibleObject` to `ListViewItem` in `List`, `Tile`, `SmallIcon`, `LagreIcon` styles? In these styles, we haven't `SubItems`, and currently, the tree looks neat:

> List1
>> ListViewItem1
>> ListViewItem2
>> ListViewItem3

But in my opinion, if we will rework it like `Details` style it looks cumbersome (especially for `ListViewItem` without Image):
> List1
>> ListViewItem1
>>> ListViewSubItem1

>> ListViewItem2
>>> ListViewImage
>>> ListViewSubItem1

>> ListViewItem3
>>> ListViewSubItem1

In all of these styles, a `ListViewItem` cannot contain more than one `SubItem`. @merriemcgaw what do you think if we add child `AccessibleObject` only for `Image` like a `Subtitle` (see image below)?
![image](https://user-images.githubusercontent.com/109065597/193571186-cf6a3a98-8565-469d-acb4-7e251cd8ed4f.png)

**Comment by merriemcgaw** [comment]
It sounds reasonable to me but also want @Tanya-Solyanik's input here too.

**Comment by Tanya-Solyanik** [comment]
@NikitaSemenovAkvelon  - I did not understand your suggestion. you are saying that this:

>List1
>>ListViewItem1
>>>ListViewSubItem1`

adds hierarchy where none is needed, right? But what is the alternative? No subitem?

>ListView1
>>ListViewItem2
>>>ListViewItemImage

And a different question - is it possible to set accessible name on this image?

**Comment by NikitaSemenovAkvelon** [comment]
@Tanya-Solyanik exactly. `ListViewItemAccessibleObject` will have only children like an image or subtitle without subItem. I attached the image above with an example of how it works now with subtitles. Item1 has a subtitle, subsequently, it has a child AO, but Item2 hasn't a subtitle therefore it without any children.

**Comment by Tanya-Solyanik** [comment]
>I don't see any easy way to set it without adding a new property to ListViewItem which will be used in the AO.

This is a problem. There is no point in announcing "ListVIewItem2, image" because the user would not know what that image is. It's valuable to have image node in the hierarchy only when that image had text associated with it. But let's wait until this issue is raised by external users.

I think hierarchy with no mandatory subitems is reasonable.

**Comment by NikitaSemenovAkvelon** [comment]
Tested the remaining cases which are still unchecked.

- Details.Case 1. OK
![Screenshot 2022-10-31 160430](https://user-images.githubusercontent.com/109065597/199005367-f9a30655-3d8e-45c7-bfa2-9b65e00be9ef.png)

- Details.Case 3. OK
![Screenshot 2022-10-31 161012](https://user-images.githubusercontent.com/109065597/199005467-4f453f93-10b8-4e1d-abda-3b47a2c1f9ad.png)

- Tile.Case 2. OK
![Screenshot 2022-10-31 161201](https://user-images.githubusercontent.com/109065597/199005583-89bd8f81-a075-48ad-ace6-abe891cc8917.png)

I suggest marking these cases as completed.

**Comment by NikitaSemenovAkvelon** [comment]
@merriemcgaw I think, we can close the issue, doesn't it?

**Comment by MandiMan** [comment]
Verified on latest 8.0 build .NET 8.0.100-alpha.1.22607.6, a part of the issue was fixed. User can see information about image via accessibility tools in LargeIcon, SmallIcon and List views. please check below screenshot:
LargeIcon view:
![LargeIcon view](https://user-images.githubusercontent.com/108860782/206401384-e28335b0-e12f-4ce4-8f80-6f4d2ffa854c.png)
SmallIcon view:
![SmallIcon view](https://user-images.githubusercontent.com/108860782/206401487-ff69b3e8-0f35-44ea-bc98-88f0ac17182f.png)
List view:
![List view](https://user-images.githubusercontent.com/108860782/206401647-f9622fad-8966-4ce9-a8d6-a1fbc22d74db.png)
